drivers/gles2/rasterizer_scene_gles2.cpp

```diff
@@ -1385,21 +1385,241 @@ bool RasterizerSceneGLES2::_setup_material(RasterizerStorageGLES2::Material *p_m
+	RasterizerStorageGLES2::Surface *s = static_cast<RasterizerStorageGLES2::Surface *>(p_element->geometry);
+	if (!s->blend_shape_data.empty()) {
+		if (transform_buffer.size() < s->array_byte_size) {
+			transform_buffer.resize(s->array_byte_size);
+		}
+		for (int i = 0; i < VS::ARRAY_MAX - 1; i++) {
+			if (s->attribs[i].enabled) {
+				const float *p_weights = p_element->instance->blend_values.ptr();
+
+				PoolVector<float>::Write write = transform_buffer.write();
+				PoolVector<uint8_t>::Read read = s->data.read();
+				float attrib_array[4] = { 0.0 };
+
+				// Read all attributes
+				for (int j = 0; j < s->array_len; j++) {
+					size_t offset = s->attribs[i].offset + (j * s->attribs[i].stride);
+					float base_weight = 1.0;
+
+					if (s->mesh->blend_shape_mode == VS::BLEND_SHAPE_MODE_NORMALIZED) {
+						for (int ti = 0; ti < s->blend_shape_data.size(); ti++) {
+							base_weight -= p_weights[ti];
+						}
+					}
+
+					// Set the base
+					switch (i) {
+						case VS::ARRAY_VERTEX: {
+							if (s->format & VS::ARRAY_COMPRESS_VERTEX) {
+								const uint16_t *v = (const uint16_t *)(read.ptr() + offset);
+								attrib_array[0] = Math::halfptr_to_float(&v[0]) * base_weight;
+								attrib_array[1] = Math::halfptr_to_float(&v[1]) * base_weight;
+								attrib_array[2] = Math::halfptr_to_float(&v[2]) * base_weight;
+							} else {
+								const float *v = (const float *)(read.ptr() + offset);
+								attrib_array[0] = v[0] * base_weight;
+								attrib_array[1] = v[1] * base_weight;
+								attrib_array[2] = v[2] * base_weight;
+							}
+						} break;
+						case VS::ARRAY_NORMAL: {
+							if (s->format & VS::ARRAY_COMPRESS_NORMAL) {
+								const float multiplier = 1.f / 127.f;
+								const int8_t *v = (const int8_t *)(read.ptr() + offset);
+								attrib_array[0] = (float(v[0]) * multiplier) * base_weight;
+								attrib_array[1] = (float(v[1]) * multiplier) * base_weight;
+								attrib_array[2] = (float(v[2]) * multiplier) * base_weight;
+							} else {
+								const float *v = (const float *)(read.ptr() + offset);
+								attrib_array[0] = v[0] * base_weight;
+								attrib_array[1] = v[1] * base_weight;
+								attrib_array[2] = v[2] * base_weight;
+							}
+						} break;
+						case VS::ARRAY_TANGENT: {
+							if (s->format & VS::ARRAY_COMPRESS_TANGENT) {
+								const float multiplier = 1.f / 127.f;
+								const int8_t *v = (const int8_t *)(read.ptr() + offset);
+								attrib_array[0] = (float(v[0]) * multiplier) * base_weight;
+								attrib_array[1] = (float(v[1]) * multiplier) * base_weight;
+								attrib_array[2] = (float(v[2]) * multiplier) * base_weight;
+							} else {
+								const float *v = (const float *)(read.ptr() + offset);
+								attrib_array[0] = v[0] * base_weight;
+								attrib_array[1] = v[1] * base_weight;
+								attrib_array[2] = v[2] * base_weight;
+							}
+						} break;
```

Comment:
Tangents have 4 components, the fourth one can just be copied over
https://github.com/godotengine/godot/blob/f34a8c7f0e9d127acc95ccff6839c7667ab736c0/drivers/gles3/shaders/blend_shape.glsl#L156-L160